abi/cbor_bytes.go

```diff
@@ -0,0 +1,51 @@
+package abi
+
+import (
+	"fmt"
+	"io"
+
+	cbg "github.com/whyrusleeping/cbor-gen"
+	"golang.org/x/xerrors"
+)
+
+type CborBytes []byte
+
+func (t *CborBytes) MarshalCBOR(w io.Writer) error {
+	if len(*t) > cbg.ByteArrayMaxLen {
+		return xerrors.Errorf("byte array was too long")
+	}
+
+	if err := cbg.WriteMajorTypeHeader(w, cbg.MajByteString, uint64(len(*t))); err != nil {
+		return err
+	}
+
+	_, err := w.Write((*t)[:])
+	return err
+}
+
+func (t *CborBytes) UnmarshalCBOR(r io.Reader) error {
+
+	br := cbg.GetPeeker(r)
+	maj, extra, err := cbg.CborReadHeader(br)
+	if err != nil {
+		return err
+	}
+
+	if extra > cbg.ByteArrayMaxLen {
```

Comment:
This is probably too small (e.g., it won't fit an entire smart contract).

Reply:
Hmm, if so we have a more serious problem -- this limit applies to pretty much every type in Filecoin (as implemented by Golang clients), including the `Message` itself. That means Lotus won't consider such messages valid.

(what happens here is less important since it's only used for UX)

Reply:
Steven is confused. Steven will go back to the caverns of rust.

(the max here is ~2MiB and isn't a problem)